src/dispatch/batch.rs

```diff
@@ -0,0 +1,418 @@
+use crate::{
+    dispatch::Dispatcher, world::ResourceId, Accessor, AccessorCow, DynamicSystemData, RunningTime,
+    System, SystemData, World,
+};
+
+/// The `BatchAccessor` is used to notify the main dispatcher of the read and
+/// write resources of the `System`s contained by the `Batch` ("sub systems").
+#[derive(Debug)]
+pub struct BatchAccessor {
```

Comment:
This seems more generic that just batch related: It just holds `reads` and `writes` that cannot be changed inside it. Not sure if it's worth renaming.

Reply:
Well currently it's not *that* dynamic as those fields are private and cannot thus be edited dynamically. I don't have any good names for it either.

Reply:
It's something that is tight with the `Batch` execution. Since it seems the most descriptive name I'm leaving it as is until we don't have a better idea.